extensions/cache/src/main/java/org/radargun/stages/cache/test/legacy/CacheTestStage.java

```diff
@@ -34,11 +35,38 @@
       complexConverter = CacheSelector.ComplexConverter.class)
    protected CacheSelector cacheSelector = null;
 
+   @Property(doc = "Specifies whether the test should fail in case the cluster size changes during the test. Default is true.")
+   public boolean failIfClusterSizeChanges = true;
+
    @InjectTrait
    protected ConditionalOperations conditionalOperations;
 
+   @InjectTrait
+   protected Clustered clustered;
+
+   protected int clusterSizeAtTestStart = 0;
+   protected int membershipHistorySizeAtStart = 0;
+
+   @Override
+   protected void destroy() {
+      if (failIfClusterSizeChanges && clustered != null) {
+         for (int i = membershipHistorySizeAtStart; i < clustered.getMembershipHistory().size(); i++) {
+            int clusterSizeDuringTest = clustered.getMembershipHistory().get(i).members.size();
+            if (clusterSizeDuringTest != clusterSizeAtTestStart) {
+               throw new RuntimeException("Cluster size at the beginning of the test was " + clusterSizeAtTestStart + " but changed to " + clusterSizeDuringTest +
```

Comment:
Now it occured ot me that maybe it would be enough to check if the size of getMembershipHistory() changed.